feat: add description property and ut for systems endpoint

amarnath-ac opened the pull request:
- Add description property to systems endpoint
- Implemented unit test for get systems endpoint
- Achieve 100% code coverage for GetRedfishV1Systems function

amarnath-ac committed:
feat: add description property and ut for systems endpoint

- Add description property to systems endpoint
- Implemented unit test for get systems endpoint
- Achieve 100% code coverage for GetRedfishV1Systems function

Signed-off-by: C, Amarnath <amarnath.c@intel.com>

sudhir-intc commented on redfish/internal/controller/http/v1/handler/routes.go:
```diff
@@ -105,7 +115,7 @@ func (s *RedfishServer) GetRedfishV1Systems(c *gin.Context) {
 		OdataType:         StringPtr("#ComputerSystemCollection.ComputerSystemCollection"),
 		Name:              "Computer System Collection",
-		Description:       nil,
+		Description:       CreateDescription("Collection of Computer Systems"),
```
As part of this implementation, please consider extracting all the hardcoded strings to constants, as it makes it easier to maintain and update URLs/metadata + reduces duplication.

For ex:
```
const (
	// Task state constants from Redfish Task.v1_8_0 specification
	taskStateCompleted = "Completed"

	// Registry message IDs
	msgIDBaseSuccess      = "Base.1.22.0.Success"
	msgIDBaseGeneralError = "Base.1.22.0.GeneralError"

	// OData metadata constants
	odataContextSystems        = "/redfish/v1/$metadata#ComputerSystemCollection.ComputerSystemCollection"
	odataIDSystems             = "/redfish/v1/Systems"
	odataTypeSystemsCollection = "#ComputerSystemCollection.ComputerSystemCollection"
	systemsCollectionName      = "Computer System Collection"
	systemsCollectionDesc      = "Collection of Computer Systems"
)
```

In this function update to
```
collection := generated.ComputerSystemCollectionComputerSystemCollection{
		OdataContext:      StringPtr(odataContextSystems),
		OdataId:           StringPtr(odataIDSystems),
		OdataType:         StringPtr(odataTypeSystemsCollection),
		Name:              systemsCollectionName,
		Description:       CreateDescription(systemsCollectionDesc),
		MembersOdataCount: Int64Ptr(int64(len(members))),
		Members:           &members,
	}
```

amarnath-ac replied:
fixed

sudhir-intc reviewed (commented):
How about the support for the following, are you planning to implement them in a different PR ?
1. Pagination Support when we need to support a larger list of systems(skip/top + nextLink)
2. Ordering/Filtering support ( ex: ?$filter=Status/Health eq 'Ok' , ?$orderby=Name asc)

refer : https://www.dmtf.org/sites/default/files/standards/documents/DSP0266_1.23.0.html#query-parameters

amarnath-ac commented:
> How about the support for the following, are you planning to implement them in a different PR ?
> 
> 1. Pagination Support when we need to support a larger list of systems(skip/top + nextLink)
> 2. Ordering/Filtering support ( ex: ?$filter=Status/Health eq 'Ok' , ?$orderby=Name asc)
> 
> refer : https://www.dmtf.org/sites/default/files/standards/documents/DSP0266_1.23.0.html#query-parameters

Will implement in separate task, created jira story : https://jira.devtools.intel.com/browse/CM-103

amarnath-ac commented on redfish/internal/controller/http/v1/handler/routes.go:
```diff
@@ -51,6 +51,16 @@ func Int64Ptr(i int64) *int64 {
 	return &i
 }
 
+// CreateDescription creates a Description from a string using ResourceDescription.
+func CreateDescription(desc string) *generated.ComputerSystemCollectionComputerSystemCollection_Description {
+	description := &generated.ComputerSystemCollectionComputerSystemCollection_Description{}
+	if err := description.FromResourceDescription(desc); err != nil {
+		return nil
+	}
+
+	return description
+}
```
fixed

amarnath-ac committed:
refactor: Extract hardcoded strings to constants and improve error logging

- Extract all hardcoded OData metadata strings to organized constants
- Add proper error logging to CreateDescription function with logger parameter
